module/metrics.go

```diff
@@ -918,6 +919,20 @@ type AccessMetrics interface {
 	UpdateLastFullBlockHeight(height uint64)
 }
 
+type RegisterDBPrunerMetrics interface {
+	// Pruned tracks the last pruned height and the pruning operation duration
+	Pruned(height uint64, duration time.Duration)
+
+	// NumberOfBlocksPruned tracks the number of blocks that were pruned during the operation.
+	NumberOfBlocksPruned(blocks uint64)
+
+	// NumberOfRowsPruned tracks the number of rows that were pruned during the operation.
+	NumberOfRowsPruned(rows uint64)
+
+	// ElementVisited records the element that were visited during the pruning operation.
+	ElementVisited()
```

Comment:
I'm thinking about how we monitor the pruning. 

We are probably interested: 
1. The last pruned height. It's a useful metrics so that we know the script query below this height would fail. 
2. Is it pruning right now. If yes, what's the progress percentage. (e.g 5%, 50%).

We are probably not interested:
1. how many actual registers are pruned. If we are interested, we can estimated by looking at how often the last pruning progress changes, since pruning is done by batch delete, and each batch has the same size.

`NumberOfRowsPruned` tells us it's pruning, but it doesn't tell us the progress. 
`ElementVisited` also tell su it's pruning, but the actual number is not very meaningful. 

I think we could consider just measure one metrics:`LatestPrunedHeightWithProgressPercentage`, it could be just a uint64 value. 

So if the metrics shows `8923910015`, then it means it's pruning, and the last pruned height `89239100` and the progress is `15%` in the existing pruning iteration. Once the current pruning iteration is completed, the metrics will become `8924910000`, which means we pruned from `89239100` to `89249100` (100%).

Now the question is, how do we know the 15% pruning progress? 

We can estimate that by checking the first few hex chars in the register ID key, since we are iterate all keys in a certain order and assuming the keys are distributed evenly, the first few hex chars basically divides all registers into different buckets, and we can calculate the percentage from that.

With only one single metrics, we could reduce the impact to the key iteration.